docs/setup-cluster/k8s/internal-task-gateway.rst

```diff
@@ -0,0 +1,123 @@
+
+In these docs, we'll be using `Contour from Project Contour <https://projectcontour.io/>`_.
+
+############################
+ Sample Setup - Development
+############################
+
+For internal testing and development, we provide a simple setup script that uses a dynamic
+provisioner provided by Project Contour.
+
+On a local dev machine, you can use Minikube and Contour as the controller. We provide a script to
+simplify the process. This can be found in `tools/k8s/launch-minikube-with-gateway.sh`.
+
+After you have a working K8s cluster and a Gateway controller running, configure the resource
+manager via master config and start the Determined cluster.
+
+###############
+ Configuration
+###############
+
+Below you'll find details on how to configure your cluster and Determined master to use the Internal
+Task Gateway.
+
+-  Total active proxies will be limited by: maxItems set in the Gateway CRD and the portRange
+   configured for Determined (not exhaustive).
+
+**********************
+ Master Configuration
+**********************
+
+To configure the optional InternalTaskGateway for a K8s resource manager, you need to add a struct
+under `internal_task_gateway` key under each of the desired resource manager configurations.
+
+This is represented by Go package `config.InternalTaskGatewayConfig` defined in
+`master/internal/config/resource_manager_config.go`
+
+.. code:: go
+
+   // InternalTaskGatewayConfig is config for exposing Determined tasks to outside of the cluster.
+   // Useful for multirm when we can only be running in a single cluster.
+   type InternalTaskGatewayConfig struct {
+       // GatewayName as defined in the k8s cluster.
+       GatewayName string `json:"gateway_name"`
+       // GatewayNamespace as defined in the k8s cluster.
+       GatewayNamespace string `json:"gateway_namespace"`
+       GatewayIP        string `json:"gateway_ip"`
+       // GWPortStart denotes the inclusive start of the available and exclusive port range to
+       // MLDE for InternalTaskGateway.
+       GWPortStart int `json:"gateway_port_range_start"`
+       // GWPortEnd denotes the inclusive end of the available and exclusive port range to
+       // MLDE for InternalTaskGateway.
+       GWPortEnd int `json:"gateway_port_range_end"`
+   }
+
+Note that the valid port range starts from 1025 to 65535, inclusive. - CHECK: might want to set max
+aux containers < min(this and port range)
+
+*********
+ Gateway
+*********
+
+In the CRD `gateways.gateway.networking.k8s.io`
+`schema.openAPIV3Schema.properties.spec.properties.listeners.maxItems` defines a max limit of how
+many listeners can be active on a single gateway. This limit sets the upper bound on how many tasks
+can be actively proxied.
+
+Note that when configuring this number you might hit K8s validation complexity thresholds checks.
+This can be configured and is dependent on each K8s cluster's requirements and setup.
```

Comment:
how would a user change this to 128?

Reply:
only the cluster admin can, basically something similar to https://github.com/determined-ai/determined/pull/9474/commits/b291343b7c0bd3763750a9256629e6e395fb2271 I can document that.